src/File/Upload.php

```diff
@@ -164,9 +182,15 @@ public function isValid($value, $file = null)
 
         foreach ($files as $file => $content) {
             $this->value = $file;
-            switch ($content['error']) {
+            $error = ($content instanceof UploadedFileInterface) ?
+                $content->getError() : $content['error'];
+            switch ($error) {
                 case 0:
-                    if (! is_uploaded_file($content['tmp_name'])) {
+                    $tmpFile = ($content instanceof UploadedFileInterface) ?
+                        $content->getStream()->getMetadata('uri') :
+                        $content['tmp_name'];
+
+                    if (! is_uploaded_file($tmpFile)) {
```

Comment:
The above logic is incorrect.

For PSR-7, we can assume that if we have UPLOAD_ERROR_OK, then the upload performed correctly. Why? because `tmp_name` has been taken out of the equation, and we're working with streams instead.

I'll update this during merge.